code/02-nix-commands/02-use-packages-in-nix-shell.md

```diff
@@ -23,4 +22,8 @@ $ nix-shell -p nodejs ghc cabal-install
 
 [nix-shell:nix-workshop]$ which cabal
 /nix/store/060x141b9fz2pm6yz4zn3i0ncavbdbf7-cabal-install-3.2.0.0/bin/cabal
-```
\ No newline at end of file
+```
+
+## Using `shell.nix`
+
+`nix-shell` will read `shell.nix` if it exists, so if you ever see a `shell.nix` in a project, know that you can run `nix-shell` to get a temporary environment with the tools you'll need. Managing temporary environments --- for any tool, written in any language --- is one of Nix's killer features.
```

Comment:
Saying something is a killer feature is not a convincing enough argument for the readers. Perhaps something like:

> Other than explicitly specifying the package dependencies in the `-p` command line option, it is also common practice for Nix projects to define a `shell.nix` file that specifies the shell environment. This is a useful feature, as we can use Nix to provide any tool, written in any language, inside the shell environment without polluting the global system. We will cover the use of `shell.nix` in a [later chapter](../04-derivations/04-raw-derivation.md#nix-shell)